julee_example/services/knowledge_service/anthropic/knowledge_service.py

```diff
@@ -89,23 +90,30 @@ async def register_file(
         logger.debug(
             "Registering file with Anthropic",
             extra={
-                "knowledge_service_id": self.config.knowledge_service_id,
+                "knowledge_service_id": config.knowledge_service_id,
                 "document_id": document.document_id,
             },
         )
 
         try:
+            # Get Anthropic client for this operation
+            client = self._get_client(config)
 
-            # Reset stream position and pass stream to Anthropic
-            assert document.content is not None
-            document.content.seek(0)
+            # Ensure content stream is positioned at beginning for upload
+            if document.content:
+                document.content.seek(0)
 
             # Upload file using Anthropic beta Files API
             # Use tuple format: (filename, file_stream, media_type)
-            file_response = await self.client.beta.files.upload(
+            if not document.content:
+                raise ValueError(
+                    "Document content stream is required for upload"
+                )
+
+            file_response = await client.beta.files.upload(
                 file=(
                     document.original_filename,
-                    cast(IO[bytes], document.content.stream),
+                    document.content.stream,  # type: ignore[arg-type]
```

Comment:
THIS took me a long time to find as the cause of an actual error - even though it was just a typing change to satisfy mypy, it actually broke the implementation (no content was actually uploaded, but no error either, just empty files in buckets). Hence switching back to document.content.stream.